api/api.go

```diff
@@ -24,3 +28,64 @@ func New(baseUrl, apiKey string, useragent string, client *http.Client) *API {
 		client: client,
 	}
 }
+
+type RetryRequest struct {
+	FunctionName string
+	ResourceName string
+	Attempt      int
+	Sleep        time.Duration
+	Data         any
+	Failed       *map[string]any
+}
+
+func CallWithRetry(ctx context.Context, sling *sling.Sling, request RetryRequest) error {
```

Comment:
```go
func (*API) callWithRetry(ctx context.Context, sling *sling.Sling, request RetryRequest) error {
```

Does it make sense to have this as a private method on the API and have every internal method call this? So that it looks something like this:
```go
api.ReadCloudAMQPInstances(ctx,....) {
  ....
  api.callWithRetry(ctx, sling, retryData) 
}
```